infra/enrolment-records-store/src/main/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSource.kt

```diff
@@ -0,0 +1,163 @@
+import com.simprints.infra.enrolment.records.store.domain.models.SubjectQuery
+import com.simprints.infra.events.event.cosync.CoSyncEnrolmentRecordEvents
+import com.simprints.infra.events.event.domain.models.subject.EnrolmentRecordCreationEvent
+import com.simprints.infra.events.event.domain.models.subject.FaceReference
+import com.simprints.infra.events.event.domain.models.subject.FingerprintReference
+import com.simprints.infra.logging.Simber
+import com.simprints.libsimprints.Constants.SIMPRINTS_COSYNC_SUBJECT_ACTIONS
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class CommCareIdentityDataSource @Inject constructor(
+    private val encoder: EncodingUtils,
+    private val jsonHelper: JsonHelper,
+    @ApplicationContext private val context: Context,
+) : IdentityDataSource {
+
+    companion object {
+        val CASE_METADATA_URI: Uri = Uri.parse("content://org.commcare.dalvik.case/casedb/case")
+        val CASE_DATA_URI: Uri = Uri.parse("content://org.commcare.dalvik.case/casedb/data")
+        const val COLUMN_CASE_ID = "case_id"
+        const val COLUMN_DATUM_ID = "datum_id"
+        const val COLUMN_VALUE = "value"
+    }
+
+    override suspend fun loadFingerprintIdentities(
+        query: SubjectQuery,
+        range: IntRange,
+    ): List<FingerprintIdentity> = loadEnrolmentRecordCreationEvents(range).map {
+        FingerprintIdentity(
+            it.payload.subjectId,
+            it.payload.biometricReferences.filterIsInstance<FingerprintReference>()
+            .flatMap { fingerprintReference ->
+                fingerprintReference.templates.map { fingerprintTemplate ->
+                    FingerprintSample(
+                        fingerIdentifier = fingerprintTemplate.finger,
+                        templateQualityScore = fingerprintTemplate.quality,
+                        template = encoder.base64ToBytes(fingerprintTemplate.template),
+                        format = fingerprintReference.format,
+                    )
+                }
+            }
+        )
+    }
+
+    override suspend fun loadFaceIdentities(
+        query: SubjectQuery,
+        range: IntRange,
+    ): List<FaceIdentity> = loadEnrolmentRecordCreationEvents(range).map {
+        FaceIdentity(
+            it.payload.subjectId,
+            it.payload.biometricReferences.filterIsInstance<FaceReference>()
+            .flatMap { faceReference ->
+                faceReference.templates.map { faceTemplate ->
+                    FaceSample(
+                        template = encoder.base64ToBytes(faceTemplate.template),
+                        format = faceReference.format,
+                    )
+                }
+            }
+        )
+    }
+
+    private fun loadEnrolmentRecordCreationEvents(range: IntRange): List<EnrolmentRecordCreationEvent> {
+        val enrolmentRecordCreationEvents: MutableList<EnrolmentRecordCreationEvent> = mutableListOf()
```

Comment:
Would be better to use [Kotlin's use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) for cursor operations to ensure the proper closure of the resources once done or exception is caught. Currently, the cursor isn't closed when the exception is thrown.

```
try {
    context.contentResolver.query(CASE_METADATA_URI, null, null, null, null).use {
        if (caseMetadataCursor != null && caseMetadataCursor.moveToPosition(range.first)) { ... }
    }
} catch { ... }
```